File: server/nit-colombia.js
/**
 * NIT Colombia: base 9–10 dígitos (puntos opcionales), guion, DV de un dígito.
 * Ej.: "890.903.938 - 8" → { ok: true, nit: "890903938-8" }
 */

function validarYNormalizarNitPh(input) {
  const raw = String(input ?? '').trim();
  if (!raw) {
    return { ok: false, error: 'El NIT / identificación de la propiedad horizontal es obligatorio.', nit: null };
  }

  const partes = raw.split(/\s*-\s*/);
  if (partes.length !== 2) {
    return {
      ok: false,
      error: 'Indica el guion entre la base y el DV (ejemplo: 890.903.938 - 8).',
      nit: null,
    };
  }

  const baseDigitos = partes[0].replace(/[.\s]/g, '');
  const dv = partes[1].replace(/\s/g, '');

  if (!/^\d+$/.test(baseDigitos)) {
    return {
      ok: false,
      error: 'La base del NIT solo debe contener dígitos (puede usar puntos como separadores).',
      nit: null,
    };
  }
  if (baseDigitos.length < 9 || baseDigitos.length > 10) {
    return {
      ok: false,
      error: 'La base del NIT debe tener entre 9 y 10 dígitos (sin contar el dígito de verificación).',
      nit: null,
    };
  }
  if (!/^\d$/.test(dv)) {
    return {
      ok: false,
      error: 'El dígito de verificación (DV) debe ser un solo número (0-9).',
      nit: null,
    };
  }

  return { ok: true, error: null, nit: `${baseDigitos}-${dv}` };
}

module.exports = { validarYNormalizarNitPh };

File: torreal-api/src/main/java/com/torreal/api/util/NitColombia.java
package com.torreal.api.util;

import java.util.regex.Pattern;

public final class NitColombia {

  private static final Pattern DIGITS = Pattern.compile("^\\d+$");
  private static final Pattern SINGLE_DV = Pattern.compile("^\\d$");

  private NitColombia() {}

  public record NitResult(boolean ok, String error, String nit) {}

  public static NitResult validarYNormalizarNitPh(String input) {
    String raw = input == null ? "" : input.trim();
    if (raw.isEmpty()) {
      return new NitResult(
          false, "El NIT / identificación de la propiedad horizontal es obligatorio.", null);
    }
    String[] partes = raw.split("\\s*-\\s*");
    if (partes.length != 2) {
      return new NitResult(
          false, "Indica el guion entre la base y el DV (ejemplo: 890.903.938 - 8).", null);
    }
    String baseDigitos = partes[0].replaceAll("[.\\s]", "");
    String dv = partes[1].replaceAll("\\s", "");
    if (!DIGITS.matcher(baseDigitos).matches()) {
      return new NitResult(
          false,
          "La base del NIT solo debe contener dígitos (puede usar puntos como separadores).",
          null);
    }
    if (baseDigitos.length() < 9 || baseDigitos.length() > 10) {
      return new NitResult(
          false,
          "La base del NIT debe tener entre 9 y 10 dígitos (sin contar el dígito de verificación).",
          null);
    }
    if (!SINGLE_DV.matcher(dv).matches()) {
      return new NitResult(
          false, "El dígito de verificación (DV) debe ser un solo número (0-9).", null);
    }
    return new NitResult(true, null, baseDigitos + "-" + dv);
  }
}
